### DISCUSSION TITLE
snowflake_warehouse RESOURCE_CONSTRAINT parameter

### DISCUSSION THREAD

**Discussion by PatriciaAyuso** [Snowflake-Labs/terraform-provider-snowflake#3241] [Q&A] [answered]
Hello! 
We're using a Snowpark Optimized warehouse and we need to modify its RESOURCE_CONSTRAINT.
Does anyone know how to do it using this TF library?
Thanks!

**Comment by sfc-gh-asawicki** [comment, marked as answer]
Hey @PatriciaAyuso. Thanks for reaching out to us.

AFAIK, resource constraints are currently in public preview (https://docs.snowflake.com/en/user-guide/warehouses-snowpark-optimized#creating-a-snowpark-optimized-warehouse) and were added after our warehouse rework before the V1. We have them in our internal backlog but we won't address them before the V1 of the provider.

Currently, you can utilize https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/unsafe_execute to run any arbitrary SQL (in this case altering warehouse to set the desired constraint after creation).

**Reply by PatriciaAyuso** [reply]
Thanks @sfc-gh-asawicki !
We're using an older library version that doesn't have the unsafe_execute resource, so I managed to do it using https://registry.terraform.io/providers/aidanmelen/snowsql/latest/docs/resources/exec.

**Reply by sfc-gh-asawicki** [reply]
Hey @PatriciaAyuso. Which version do you use? I would strongly recommend migrating to the newer versions as they are much more stable and offer more features. :)
